hooks/_common.sh

```diff
@@ -445,13 +447,66 @@ function common::colorify {
   echo -e "${COLOR}${TEXT}${RESET}" >&2
 }
 
+#######################################################################
+# Get Terraform/OpenTofu binary path
+# Allows user to set the path to custom Terraform or OpenTofu binary
+# Globals (init and populate):
+#   HOOK_CONFIG (array) arguments that configure hook behavior
+#   PCT_TFPATH (string) user defined env var with path to Terraform/OpenTofu binary
+#   TERRAGRUNT_TFPATH (string) user defined env var with path to Terraform/OpenTofu binary
+# Outputs:
+#   If failed - exit 1 with error message about missing Terraform/OpenTofu binary
+#######################################################################
+function common::get_tf_binary_path {
+  local hook_config_tf_path
+
+  for config in "${HOOK_CONFIG[@]}"; do
+    if [[ $config == --tf-path=* ]]; then
+      hook_config_tf_path=${config#*=}
+      hook_config_tf_path=${hook_config_tf_path%;}
+      break
+    fi
+  done
+
+  # direct hook config, has the highest precedence
+  if [[ $hook_config_tf_path ]]; then
+    echo "${hook_config_tf_path}"
+    return
+
+  # environment variable
+  elif [[ $PCT_TFPATH ]]; then
+    echo "${PCT_TFPATH}"
+    return
+
+  # Maybe there is a similar setting for Terragrunt already
+  elif [[ $TERRAGRUNT_TFPATH ]]; then
+    echo "${TERRAGRUNT_TFPATH}"
+    return
+
+  # check if Terraform binary is available
+  elif command -v terraform &> /dev/null; then
+    command -v terraform
+    return
+
+  # finally, check if Tofu binary is available
+  elif command -v tofu &> /dev/null; then
+    command -v tofu
+    return
+
+  else
+    common::colorify "red" "Neither Terraform nor OpenTofu binary could be found. Please either set the \"--tf-path\" hook configuration argument, or set the \"PCT_TFPATH\" environment variable, or set the \"TERRAGRUNT_TFPATH\" environment variable, or install Terraform or OpenTofu globally."
+    exit 1
```

Comment:
```suggestion
    common::colorify "yellow" "Neither Terraform nor OpenTofu binary could be found. If hook crashed, please either set the '--tf-path' hook configuration argument, or set the 'PCT_TFPATH' environment variable, or set the 'TERRAGRUNT_TFPATH' environment variable, or install Terraform or OpenTofu globally."
```

We have hooks that don't need tf binaries, so it can be not error in some cases

Reply:
@MaxymVlasov Hope you don't mind that I replaced `crashed` with `fails` in your commit suggestion 🤞🏻 
I also replaced `globally` with `so it is available in the 'PATH' environment variable` to make it clear that binary is looked up from the `PATH`.